### PR TITLE
feat: Use nightly olm channel.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What does this PR do?
Use nightly olm channel.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17183

### Depends on:
https://github.com/eclipse/che-operator/pull/423

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

File: src/commands/server/start.ts [hunk 1]
```diff
@@ -441,7 +440,7 @@ export default class Start extends Command {
       return
     }
 
-    if (flags.platform === 'openshift' && await kubeHelper.isOpenShift4() && isStableVersion(flags) && olmIsPreinstalled) {
+    if (flags.platform === 'openshift' && await kubeHelper.isOpenShift4() && olmIsPreinstalled) {
```
Comment (mmorhun) [inline src/commands/server/start.ts:443]: `olmIsPreinstalled` -> `IsOlmPreinstalled`

**Review by mmorhun — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -461,10 +461,10 @@ OPTIONS
                                            be deployed
 
   --access-token=access-token              Eclipse Che OIDC Access Token. See the documentation how to obtain token:
-                                           https://www.eclipse.org/che/docs/che-7/authenticating-users/#obtaining-the-to
```
Comment (tolusha) [inline README.md:464]: Pls. rebase on master to get the latest README.md file

File: src/tasks/installers/olm.ts [hunk 3]
```diff
@@ -53,10 +50,25 @@ export class OLMTasks {
...
 
           task.title = `${task.title}...done.`
         }
       },
+      {
+        enabled: () => !isStableVersion(flags) && !flags['catalog-source-yaml'] && !flags['catalog-source-name'],
+        title: `Create nightly index CatalogSource in the namespace ${flags.chenamespace}`,
+        task: async (ctx: any, task: any) => {
+          if (!await kube.catalogSourceExists(NIGHTLY_CATALOG_SOURCE_NAME, flags.chenamespace)) {
+            const catalogSourceImage = `quay.io/aandriienko/eclipse-che-${ctx.generalPlatformName}-opm-catalog:preview`
```
Comment (tolusha) [inline src/tasks/installers/olm.ts:63]: Should we change the image?

File: src/tasks/installers/olm.ts [hunk 4]
```diff
@@ -279,4 +299,24 @@ export class OLMTasks {
       }
     }
   }
+
+  private createIndexCatalogSource(namespace: string, catalogSourceImage: string): CatalogSource {
```
Comment (tolusha) [inline src/tasks/installers/olm.ts:303]: this function doesn't create anything but return CatalogSource yaml

File: src/tasks/installers/olm.ts [hunk 5]
```diff
@@ -80,11 +92,17 @@ export class OLMTasks {
             task.title = `${task.title}...It already exists.`
           } else {
             let subscription: Subscription
-            if (!flags['catalog-source-yaml'] && !flags['catalog-source-name']) {
+
+            // stable Che CatalogSource
+            if (!flags['catalog-source-yaml'] && !flags['catalog-source-name'] && isStableVersion(flags)) {
```
Comment (tolusha) [inline src/tasks/installers/olm.ts:97]: We often use `!flags['catalog-source-yaml'] && !flags['catalog-source-name'] && isStableVersion(flags)`
consider moving this block to function.

File: README.md (hunk 2 above)
Comment (AndrienkoAleksandr) [reply README.md:464]: This pr is build on top of the latest commit. If I checkout to master and do 'yarn run pack-binaries', README.md file is modified. I guess master branch has outdated README.md...

File: src/tasks/installers/olm.ts (hunk 4 above)
Comment (AndrienkoAleksandr) [reply src/tasks/installers/olm.ts]: I renamed this method to `constructCatalogSource`

File: README.md (hunk 2 above)
Comment (AndrienkoAleksandr) [reply README.md:464]: And I didn't forget to make `git pull origin master`

Comment (AndrienkoAleksandr) [reply README.md:464]: Ok, I see https://github.com/che-incubator/chectl/pull/847

**Review by flacatus — APPROVED** [review]

**Pull request merged**